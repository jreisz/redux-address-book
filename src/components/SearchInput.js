import React, { Component } from 'react';
import PropTypes from 'prop-types';

export default class SearchInput extends Component {  
  render() {
    const { handleSearchInput, query } = this.props;
    return (
      <div className="">
        <div className="">
          <form  id="search-form">
            <div className="input-group">
              <input
                onChange={evt => handleSearchInput(evt)}
                value={query}
                className="input-group-field"
                type="search"
                placeholder="Search..."
                ref="searchInput"
                id="search-input-field"
                autoFocus
              />
              <div className="input-group-button">
                <button className="button" type="submit">
                  Search&nbsp;
                  <i className="fi-magnifying-glass" />
                </button>
              </div>
            </div>
          </form>
        </div>
      </div>
    );
  }
}

SearchInput.propTypes = {
  handleSearchInput: PropTypes.func.isRequired,

  query: PropTypes.string,
};

SearchInput.defaultProps = {
  query: '',
};
