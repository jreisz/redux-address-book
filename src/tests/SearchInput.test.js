import React from 'react';
import { shallow } from 'enzyme';
import './enzymeAdapter';
import SearchInput from '../components/SearchInput';

const mockHandleSearchInput = jest.fn();
const mockHandleSubmitQuery = jest.fn();
const mockProps = {
  handleSearchInput: mockHandleSearchInput,
  handleSubmitQuery: mockHandleSubmitQuery,
  query: 'Anna',
};

describe('SearchInput', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<SearchInput {...mockProps} />);
  });

  it('renders like expected (snapshot)', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('has submit function', () => {
    wrapper.find('#search-form').simulate('submit', {
      preventDefault: () => {},
      target: { value: 'some value' },
    });

    expect(mockHandleSubmitQuery).toHaveBeenCalledTimes(1);
  });

  it('handles search input field', () => {
    wrapper.find('#search-input-field').simulate('change', {
      preventDefault: () => {},
      target: { value: 'some value' },
    });

    expect(mockHandleSearchInput).toHaveBeenCalledTimes(1);
  });
});
