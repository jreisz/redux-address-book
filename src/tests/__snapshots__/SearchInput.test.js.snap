// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchInput renders like expected (snapshot) 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SearchInput
    handleSearchInput={[Function]}
    handleSubmitQuery={[Function]}
    query="Anna"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className=""
>
        <form
                id="search-form"
                onSubmit={[Function]}
        >
                <div
                        className="input-group"
                >
                        <input
                                autoFocus={true}
                                className="input-group-field"
                                id="search-input-field"
                                onChange={[Function]}
                                placeholder="Search..."
                                type="search"
                                value="Anna"
                        />
                        <div
                                className="input-group-button"
                        >
                                <button
                                        className="button"
                                        type="submit"
                                >
                                        Search 
                                        <i
                                                className="fi-magnifying-glass"
                                        />
                                </button>
                        </div>
                </div>
        </form>
</div>,
      "className": "",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <form
          id="search-form"
          onSubmit={[Function]}
>
          <div
                    className="input-group"
          >
                    <input
                              autoFocus={true}
                              className="input-group-field"
                              id="search-input-field"
                              onChange={[Function]}
                              placeholder="Search..."
                              type="search"
                              value="Anna"
                    />
                    <div
                              className="input-group-button"
                    >
                              <button
                                        className="button"
                                        type="submit"
                              >
                                        Search 
                                        <i
                                                  className="fi-magnifying-glass"
                                        />
                              </button>
                    </div>
          </div>
</form>,
        "className": "",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <div
            className="input-group"
>
            <input
                        autoFocus={true}
                        className="input-group-field"
                        id="search-input-field"
                        onChange={[Function]}
                        placeholder="Search..."
                        type="search"
                        value="Anna"
            />
            <div
                        className="input-group-button"
            >
                        <button
                                    className="button"
                                    type="submit"
                        >
                                    Search 
                                    <i
                                                className="fi-magnifying-glass"
                                    />
                        </button>
            </div>
</div>,
          "id": "search-form",
          "onSubmit": [Function],
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <input
                autoFocus={true}
                className="input-group-field"
                id="search-input-field"
                onChange={[Function]}
                placeholder="Search..."
                type="search"
                value="Anna"
/>,
              <div
                className="input-group-button"
>
                <button
                                className="button"
                                type="submit"
                >
                                Search 
                                <i
                                                className="fi-magnifying-glass"
                                />
                </button>
</div>,
            ],
            "className": "input-group",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "autoFocus": true,
                "className": "input-group-field",
                "id": "search-input-field",
                "onChange": [Function],
                "placeholder": "Search...",
                "type": "search",
                "value": "Anna",
              },
              "ref": "searchInput",
              "rendered": null,
              "type": "input",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <button
                  className="button"
                  type="submit"
>
                  Search 
                  <i
                                    className="fi-magnifying-glass"
                  />
</button>,
                "className": "input-group-button",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    "Search ",
                    <i
                      className="fi-magnifying-glass"
/>,
                  ],
                  "className": "button",
                  "type": "submit",
                },
                "ref": null,
                "rendered": Array [
                  "Search ",
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "className": "fi-magnifying-glass",
                    },
                    "ref": null,
                    "rendered": null,
                    "type": "i",
                  },
                ],
                "type": "button",
              },
              "type": "div",
            },
          ],
          "type": "div",
        },
        "type": "form",
      },
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className=""
>
          <form
                    id="search-form"
                    onSubmit={[Function]}
          >
                    <div
                              className="input-group"
                    >
                              <input
                                        autoFocus={true}
                                        className="input-group-field"
                                        id="search-input-field"
                                        onChange={[Function]}
                                        placeholder="Search..."
                                        type="search"
                                        value="Anna"
                              />
                              <div
                                        className="input-group-button"
                              >
                                        <button
                                                  className="button"
                                                  type="submit"
                                        >
                                                  Search 
                                                  <i
                                                            className="fi-magnifying-glass"
                                                  />
                                        </button>
                              </div>
                    </div>
          </form>
</div>,
        "className": "",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <form
            id="search-form"
            onSubmit={[Function]}
>
            <div
                        className="input-group"
            >
                        <input
                                    autoFocus={true}
                                    className="input-group-field"
                                    id="search-input-field"
                                    onChange={[Function]}
                                    placeholder="Search..."
                                    type="search"
                                    value="Anna"
                        />
                        <div
                                    className="input-group-button"
                        >
                                    <button
                                                className="button"
                                                type="submit"
                                    >
                                                Search 
                                                <i
                                                            className="fi-magnifying-glass"
                                                />
                                    </button>
                        </div>
            </div>
</form>,
          "className": "",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <div
              className="input-group"
>
              <input
                            autoFocus={true}
                            className="input-group-field"
                            id="search-input-field"
                            onChange={[Function]}
                            placeholder="Search..."
                            type="search"
                            value="Anna"
              />
              <div
                            className="input-group-button"
              >
                            <button
                                          className="button"
                                          type="submit"
                            >
                                          Search 
                                          <i
                                                        className="fi-magnifying-glass"
                                          />
                            </button>
              </div>
</div>,
            "id": "search-form",
            "onSubmit": [Function],
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <input
                  autoFocus={true}
                  className="input-group-field"
                  id="search-input-field"
                  onChange={[Function]}
                  placeholder="Search..."
                  type="search"
                  value="Anna"
/>,
                <div
                  className="input-group-button"
>
                  <button
                                    className="button"
                                    type="submit"
                  >
                                    Search 
                                    <i
                                                      className="fi-magnifying-glass"
                                    />
                  </button>
</div>,
              ],
              "className": "input-group",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "autoFocus": true,
                  "className": "input-group-field",
                  "id": "search-input-field",
                  "onChange": [Function],
                  "placeholder": "Search...",
                  "type": "search",
                  "value": "Anna",
                },
                "ref": "searchInput",
                "rendered": null,
                "type": "input",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <button
                    className="button"
                    type="submit"
>
                    Search 
                    <i
                                        className="fi-magnifying-glass"
                    />
</button>,
                  "className": "input-group-button",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      "Search ",
                      <i
                        className="fi-magnifying-glass"
/>,
                    ],
                    "className": "button",
                    "type": "submit",
                  },
                  "ref": null,
                  "rendered": Array [
                    "Search ",
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "className": "fi-magnifying-glass",
                      },
                      "ref": null,
                      "rendered": null,
                      "type": "i",
                    },
                  ],
                  "type": "button",
                },
                "type": "div",
              },
            ],
            "type": "div",
          },
          "type": "form",
        },
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
